enhancements/kubelet/oci-image-volume.md

```diff
@@ -0,0 +1,187 @@
+---
+title: oci-image-volume
+authors:
+  - bitoku
+reviewers:
+  - haircommander
+  - saschagrunert
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - mrunalp
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2025-05-07
+last-updated: 2025-05-07
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPNODE-2929
+  - https://issues.redhat.com/browse/OCPNODE-2575
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+# OCI Image Volume
+
+## Summary
+
+In Kubernetes 1.33, support for OCI Image Volume went to Beta.
+This enhancement is to enable this feature in OpenShift.
+See [KEP-4639: Summary](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#summary)
+
+## Motivation
+
+See [KEP-4639: Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#motivation)
+
+### User Stories
+
+See [KEP-4639: User Stories](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#user-stories-optional)
+
+### Goals
+
+- Enable ImageVolume feature gate for OpenShift.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing VolumeSource types.
+- This proposal does not address other use cases for OCI objects beyond directory sharing among containers in a pod.
+- Mounting thousands of images in a single pod.
+
+## Proposal
+
+1. Enable ImageVolume feature gate in 4.20
+
+    - This feature gate is disabled by default in Kubernetes 1.33, so we need to explicitly enable it for OCP
+
+2. Separate Graduation Approach between OCI Images and OCI Artifacts.
+
+    - **OCI Image Mount**: Promote to GA
+    - **OCI Artifact Mount**: Begin as DevPreview
```

Comment:
This is a confusing to me..

Are we including the DevPreview of OCI Artifact Mount in this KEP?

Or are we aiming for this enhancement to me ImageVolume"

Reply:
Yes, the enhancement name is the same as the original KEP, which also includes OCI images and OCI artifacts.
There's not many differences feature-wise, I'd like to include OCI artfiacts rather than creating a new one for them, but it's true it's a bit confusing.

Reply:
I’m not sure if this is a big deal but I see that this KEP will have two different feature gates that will progress differently.

ImageVolume will be used to toggle the upstream feature gate so we can GA earlier. I think this gate could be dropped once ImageVolume is turned on by default.


For OCI artifacts we are starting with dev preview with no feature gates but eventually we would have a separate feature gate. I don’t know when we can promote this to GA?

writing this out here because I think we need to cover this in this KEP if you want to combine these two features but graduate at different times.

Reply:
The graduation process is going to be what @haircommander described.
But thank you for pointing it out @kannon92. That point should be explained in this section. I'll add the explanation.

Reply:
I added some explanations

Reply:
artifacts are out of scope for KEPs, do you mean openshift enhancement? I think we can append this one with graduation criteria when it's more clear, we're still defining what artifact support looks generally

Reply:
Yea this OEP.

Reply:
I think we could update the criteria if needed after this PR's merged, and if we think we should open a new draft when we graduate it to TP or GA, we can think about it at that time.